transforms/code/proglang_select/python/src/proglang_select_transform.py

```diff
@@ -0,0 +1,172 @@
+cli_prefix = f"{shortname}_"
+lang_allowed_langs_file_key = f"{shortname}_allowed_langs_file"
+lang_lang_column_key = f"{shortname}_language_column"
+lang_allowed_languages = f"{shortname}_allowed_languages"
+lang_data_factory_key = f"{shortname}_data_factory"
+lang_output_column_key = f"{shortname}_output_column"
+lang_default_output_column = "allowed_language"
+
+
+def _get_supported_languages(lang_file: str, data_access: DataAccess, logger) -> list[str]:
+    logger.info(f"Getting supported languages from file {lang_file}")
+    lang_list, _ = data_access.get_file(lang_file)
+    l_list = lang_list.decode("utf-8").splitlines()
+    logger.info(f"Supported languages {lang_list}")
+    return l_list
+
+
+class ProgLangSelectTransform(AbstractTableTransform):
+    """ """
+
+    def __init__(self, config: dict):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, LangSelectorTransformRuntime.  If running inside the RayMutatingDriver,
+        these will be provided by that class with help from the RayMutatingDriver.
+        """
+
+        super().__init__(config)
+        from data_processing.utils import get_logger
+
+        self.logger = get_logger(__name__)
+        self.lang_column = config.get(lang_lang_column_key, "")
+        self.output_column = config.get(lang_output_column_key, lang_default_output_column)
+        # languages_include_ref = config.get(lang_allowed_languages, None)
+        # if languages_include_ref is None:
+        self.languages_include = config.get(lang_allowed_languages, None)
+        if self.languages_include is None:
+            path = config.get(lang_allowed_langs_file_key, None)
+            if path is None:
+                raise RuntimeError(f"Missing configuration value for key {lang_allowed_langs_file_key}")
+            daf = config.get(lang_data_factory_key, None)
+            data_access = daf.create_data_access()
+            self.languages_include = _get_supported_languages(
+                lang_file=path, data_access=data_access, logger=self.logger
+            )
+            #  Ray now passes the list of strings instead of a reference.
+        # else:
+        #     # This is recommended for production approach. In this case domain list is build by the
+        #     # runtime once, loaded to the object store and can be accessed by actors without additional reads
+        #     try:
+        #         logger.info(f"Loading languages to include from Ray storage under reference {languages_include_ref}")
+        #         self.languages_include = ray.get(languages_include_ref)
+        #     except Exception as e:
+        #         logger.info(f"Exception loading languages list from ray object storage {e}")
+        #         raise RuntimeError(f"exception loading from object storage for key {languages_include_ref}")
+
```

Comment:
Well this is cheating. You are removing functionality important for Ray. I would suggest to move the correct init to Ray and Python version and remove it from here

Reply:
It seems reasonable to me to put the data read by the runtime config into the config dictionary instead of ray storage. the config dictionary is passed to workers and this is then more generic.  We still avoid having each worker read the data. in the end the data is passed through the network from orchestrator to worker. it might even be faster than putting in ray storage and pulling out of ray storage to just pass it once over the network as a dictionary.

Reply:
Instead of putting it in ray object storage, i'm putting it in the config dictionary (in the ray runtime object apply_inputs() method.  This still avoid having each worker read the file, which I believe is the goal in the end.  In fact, putting it in ray object storage may be less efficient then just passing it in the already passed config dict.

Reply:
It will work fine for the small amounts of data, but can be quite expensive for the last data sizes. As for the network traffic its a wash, as calls from the orchestrator is over the network. Using Ray object storage is a well defined pattern that we are promoting. Please restore it

Reply:
I might be not clear. What I am saying is that we should allow for different implementation between pure Python and Ray. Moreover I am suggesting a clean way to do this by implementing different init between Python and Ray. We had this approach there and working for a long time. Its not jumping through hoops. Its implementing the relevant mechanism for relevant runtimes